src/powerpwn/copilot_studio/modules/deep_scan.py

```diff
@@ -471,19 +475,110 @@ def print_brand(tenant: str, timeout: int = 10):
         print(f"No brand found for tenant {tenant}")
 
 
+def run_pup_commands(existing_bots):
```

Comment:
Let's please add _docstrings_ with explicit types & documentation in the passed parameters for best practice (see existing functions).